### ISSUE TITLE
Bake failed: Error: Chart.yaml file is missing (Bitbucket repository)

### ISSUE THREAD

**Issue by sramek5** [spinnaker/spinnaker#6561]
Re-opening issue #6264 

The problem unfortunately still remains.

We upgraded spinnaker to latets version -> ```1.26.6 (Disenchantment)``` except one service = rosco.
Rosco MUST stay in version ```0.22.0-20201009170017```, because all newer version gives the error ```Exception ( Create Bake ) - Bake failed: Error: Chart.yaml file is missing``` unless we explicitly add a ```helmChartFilePath``` property to the bakeManifest stage (we use git/repo artifact).

This is a really pain in the arse. I cannot force all developers to add this variable manually to the all pipelines...
So we used the old version of the rosco until today.

Now the problem is even bigger.
With new version of helm 3 (3.7.1) we cannot deploy new charts anymore...error is for example:
```
Exception ( Create Bake )
Bake failed: Error: template: archi-slonominatim/templates/all.yaml:4:21: executing "archi-slonominatim/templates/all.yaml" at <index .Subcharts "archi-sloexporter">: error calling index: index of untyped nil

Use --debug flag to render out invalid YAML
```
The atribute ```.Subcharts``` it is not known by older helm...

So we desperately need newer rosco and have an option to specify ```helmChartFilePath``` right in the gui, so everybody can see it.

Is there please any solution or progress with that problem?
Thank you.

**Comment by sramek5** [comment]
Also mentioned in #6425 (by me again).

**Comment by CalvinTse** [comment]
Hi @sramek5! Are you saying that you can't see a text field to specify the **Helm Chart File Path** for git/repo artifacts? Can I see a screenshot of this interaction?

**Comment by sramek5** [comment]
Hi @CalvinTse! Thank you for you response.

You are very right. It should be seen in Bake stage I suppose?

This is ```Spinnaker 1.26.6``` with ```rosco 0.25.0-20210422230020```:

<img width="1432" alt="Snímek obrazovky 2021-11-06 v 6 56 17" src="https://user-images.githubusercontent.com/53052347/140599690-b9116456-042f-44e8-9952-1a3458ea1ccb.png">
<img width="1432" alt="Snímek obrazovky 2021-11-06 v 6 56 30" src="https://user-images.githubusercontent.com/53052347/140599693-a2b66a87-ab29-4859-9e77-2d190efd6bd7.png">
<img width="1432" alt="Snímek obrazovky 2021-11-06 v 6 56 38" src="https://user-images.githubusercontent.com/53052347/140599695-f1187919-0bc4-40e0-9807-5401d1e78e01.png">
<img width="1432" alt="Snímek obrazovky 2021-11-06 v 6 56 51" src="https://user-images.githubusercontent.com/53052347/140599696-a443c0c2-91f1-44db-b248-44b8f8bf4d17.png">

As you can see, there is NO field for that purpose. Deployment will pass only if I ```Edit stage as JSON``` and explicitly add ```helmChartFilePath``` (= write it manually).

Next problem is old helm. ```Rosco 0.25.0-20210422230020``` has ```Helm v.3.5.4```:
```
$ kubectl exec -t -i spin-rosco-5765bb8598-lsjfm -c rosco -- bash
bash-5.0$ helm3 version
version.BuildInfo{Version:"v3.5.4", GitCommit:"1b5edb69df3d3a08df77c9902dc17af864ff05d1", GitTreeState:"clean", GoVersion:"go1.15.11"}
```

As I mentioned above, its not compatible with our newer charts. I checked docker images and found ```us-docker.pkg.dev/spinnaker-community/docker/rosco:0.26.0-20211105170020``` with ```Helm v.3.7.1``` which is pretty much what we need.:
```
$ kubectl exec -t -i spin-rosco-7b6db84c4-cbswj -c rosco -- bash 
bash-5.0$ helm3 version
version.BuildInfo{Version:"v3.7.1", GitCommit:"1d11fcb5d3f3bf00dbe6fe31b8412839a96b3dc4", GitTreeState:"clean", GoVersion:"go1.16.9"}
```
Unfortunately the bake error still remains (no field for ```helmChartFilePath```).

Is it possible to fix it?

**Comment by CalvinTse** [comment]
@sramek5 Could you send me the response body for the request to `<gate-url>/artifacts/credentials`. You can get this request/response when you click on the **Bake (Manifest)** stage and look at the Network Requests being made.

**Comment by sramek5** [comment]
Helloo @CalvinTse :-)

Did you mean this?

URL request: ```https://spinnaker-gate.service.dev2-ict.consul/artifacts/credentials```

Response:
```[{"name":"embedded-artifact","type":"artifacts-embedded","types":["embedded/base64"]},{"name":"front50ArtifactCredentials","type":"artifacts-front50","types":["front50/pipelineTemplate"]},{"name":"docker-registry","type":"artifacts-docker","types":["docker/image"]},{"name":"kubernetes","type":"artifacts-kubernetes","types":["kubernetes/configMap","kubernetes/deployment","kubernetes/replicaSet","kubernetes/secret"]},{"name":"git","type":"git/repo","types":["git/repo"]},{"name":"custom-artifact","type":"artifacts-custom","types":["custom/object"]}]```

**Comment by CalvinTse** [comment]
@sramek5 Hello :) I have made a change to Deck to show the Helm Chart File Path.

**Comment by sramek5** [comment]
Ou great, thank you very much :-) It was really blocking us.
Do I understand correctly that it is gonna be in the new release of Spinnaker 1.28?

**Comment by CalvinTse** [comment]
Yes that is correct :)

**Comment by sramek5** [comment]
Great!
Thanks again.

**Comment by dbyron-sf** [comment]
https://github.com/spinnaker/deck/pull/9773 brings this fix into the release-1.26.x branch.

**Comment by dbyron-sf** [comment]
https://github.com/spinnaker/deck/pull/9774 brings this fix into the release-1.25.x branch.

**Comment by sramek5** [comment]
Tested and it works like a charm!
Thank you.
